src/backend/web/authentication.js

```diff
@@ -122,21 +122,22 @@ function protectWithRedirect(req, res, next) {
   passport.authenticate('saml')(req, res, next);
 }
 
+function throwCustomError(message, status) {
+  const err = Error(message);
+  err.status = status;
+
+  throw err;
+}
+
 // If user is not authenticated, return an appropriate 400 error type
-function forbidden(req, res) {
-  if (req.accepts('json')) {
```

Comment:
We still want this case left in, because I don't want to send HTML back to a REST API call.  Only do your custom error path if it isn't expecting JSON.